proxy/src/registry.rs

```diff
@@ -339,11 +361,19 @@ impl Registry {
             .client
             .get_project(project_name, org_id)
             .await?
-            .map(|project| Project {
-                name: project.name,
-                org_id: project.org_id,
-                // TODO(xla): Proper conversion of ProjectIds.
-                maybe_project_id: None,
+            .map(|project| {
+                let metadata_vec: Vec<u8> = project.metadata.into();
+                Project {
+                    name: project.name,
+                    org_id: project.org_id,
+                    maybe_project_id: if let [] = metadata_vec[..] {
```

Comment:
The [`is_empty`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.is_empty) would make this more readable.